### ISSUE TITLE
Array index out of bound when performing confident learning for object detection

### ISSUE THREAD

**Issue by paul-ang** [cleanlab/cleanlab#806]
The code below in `object_detection/rank.py` produces an IndexError when len(idx_at_least_low_probability_threshold) < len(idx_at_least_intersection_threshold) : 
```
 idx_at_least_low_probability_threshold = np.where(k_pred > low_probability_threshold)[0]
 idx_at_least_intersection_threshold = np.where(k_iou > 0)[0]
 k_similarity = k_similarity[idx_at_least_low_probability_threshold][idx_at_least_intersection_threshold]
 k_pred = k_pred[idx_at_least_low_probability_threshold][idx_at_least_intersection_threshold]
```

**Comment by ulya-tkch** [comment]
Great catch, @paul-ang! Thank you for submitting the issue, fix for the indexing error is handled here #807.

**Comment by jwmueller** [comment]
This issue should've been resolved in https://github.com/cleanlab/cleanlab/pull/807

Please re-open it if you're still experiencing issues using the updated code!
